packages/plugin-ext/src/main/node/plugin-service.ts

```diff
@@ -28,11 +28,7 @@ export class PluginApiContribution implements BackendApplicationContribution {
         });
 
         app.get('/webview/:path(*)', (req, res) => {
-            let filePath: string = req.params.path;
-            if (filePath.charAt(0) !== '/') {
-                filePath = '/' + filePath;
```

Comment:
The files cannot be transferred at the given path without it.

You can try with https://github.com/microsoft/vscode-extension-samples/tree/master/webview-sample and look for internal server errors after.

All other changes can be applied.

Reply:
what exactly is not supposed to work? I tried the webview sample and I don't see any errors or any not found requests...

Reply:
> You can try with https://github.com/microsoft/vscode-extension-samples/tree/master/webview-sample and look for internal server errors after.

How can I install this as an extension? Is it available for download somewhere? I have tried to build it locally with the following changes but got compiler errors due to `vscode`:
```diff
diff --git a/webview-sample/package.json b/webview-sample/package.json
index 176f41b..d890b8f 100644
--- a/webview-sample/package.json
+++ b/webview-sample/package.json
@@ -30,16 +30,20 @@
                ]
        },
        "scripts": {
+               "postinstall": "node ./node_modules/vscode/bin/install",
                "vscode:prepublish": "npm run compile",
                "compile": "tsc -p ./",
                "lint": "tslint -p ./",
-               "watch": "tsc -w -p ./"
+               "watch": "tsc -w -p ./",
+               "package": "vsce package"
        },
        "dependencies": {},
        "devDependencies": {
                "@types/node": "^10.5.2",
                "tslint": "^5.16.0",
                "typescript": "^3.5.1",
-               "@types/vscode": "^1.32.0"
+               "@types/vscode": "^1.32.0",
+               "vsce": "^1.64.0",
+               "vscode": "^1.1.18"
        }
 }
```

Which version of `vscode` should I use to overcome the compiler errors? I would like to reproduce the
> look for internal server errors after.

first, then the fix.

Thanks, @olexii4 and @amiramw!

Reply:
By the way, the error is swallowed when I run `npm run package` but running `npm run compile` reveals the `vscode` API issues.
```
$ npm run compile

> cat-coding@0.0.1 compile C:\Users\kittaakos\dev\vscode-extension-samples\webview-sample
> tsc -p ./

node_modules/@types/vscode/index.d.ts(17,1): error TS6200: Definitions of the following identifiers conflict with those in another file: version, Position, Range, Selection, Keyboard, Mouse, Command, Line, Block, Underline, LineThin, BlockOutline, UnderlineThin, Off, On, Relative, Default, InCenter, InCenterIfOutsideViewport, AtTop, Left, Center, Right, Full, OpenOpen, ClosedClosed, OpenClosed, ClosedOpen, ThemeColor, ThemeIcon, LF, CRLF, Uri, CancellationTokenSource, Disposable, EventEmitter, RelativePattern, GlobPattern, DocumentSelector, ProviderResult, CodeActionKind, CodeAction, CodeLens, DefinitionLink, Definition, Declaration, MarkdownString, MarkedString, Hover, Text, Read, Write, DocumentHighlight, File, Module, Namespace, Package, Class, Method, Property, Field, Constructor, Enum, Interface, Function, Variable, Constant, String, Number, Boolean, Array, Object, Key, Null, EnumMember, Struct, Event, Operator, TypeParameter, SymbolInformation, DocumentSymbol, TextEdit, WorkspaceEdit, SnippetString, ParameterInformation, SignatureInformation, SignatureHelp, Invoke, TriggerCharacter, ContentChange, Unit, Value, Keyword, Snippet, Color, Reference, Folder, CompletionItem, CompletionList, TriggerForIncompleteCompletions, DocumentLink, ColorInformation, ColorPresentation, FoldingRange, Comment, Imports, Region, CharacterPair, None, Indent, IndentOutdent, Outdent, Global, Workspace, WorkspaceFolder, Location, Error, Warning, Information, Hint, DiagnosticRelatedInformation, Unnecessary, Diagnostic, Active, Beside, One, Two, Three, Four, Five, Six, Seven, Eight, Nine, Always, Silent, Never, Shared, Dedicated, New, TaskGroup, ProcessExecution, Escape, Strong, Weak, ShellExecution, Task, taskExecutions, onDidStartTask, onDidEndTask, onDidStartTaskProcess, onDidEndTaskProcess, Unknown, Directory, SymbolicLink, FileSystemError, Changed, Created, Deleted, appName, appRoot, language, clipboard, machineId, sessionId, activeTextEditor, visibleTextEditors, onDidChangeActiveTextEditor, onDidChangeVisibleTextEditors, onDidChangeTextEditorSelection, onDidChangeTextEditorVisibleRanges, onDidChangeTextEditorOptions, onDidChangeTextEditorViewColumn, terminals, activeTerminal, onDidChangeActiveTerminal, onDidOpenTerminal, onDidCloseTerminal, state, onDidChangeWindowState, TreeItem, Collapsed, Expanded, SourceControl, Window, Notification, QuickInputButtons, Manual, AfterDelay, FocusOut, rootPath, workspaceFolders, name, onDidChangeWorkspaceFolders, textDocuments, onDidOpenTextDocument, onDidCloseTextDocument, onDidChangeTextDocument, onWillSaveTextDocument, onDidSaveTextDocument, onDidChangeConfiguration, onDidChangeDiagnostics, inputBox, DebugAdapterExecutable, DebugAdapterServer, DebugAdapterDescriptor, Breakpoint, SourceBreakpoint, FunctionBreakpoint, activeDebugSession, activeDebugConsole, breakpoints, onDidChangeActiveDebugSession, onDidStartDebugSession, onDidReceiveDebugSessionCustomEvent, onDidTerminateDebugSession, onDidChangeBreakpoints, all, onDidChange
node_modules/@types/vscode/index.d.ts(3021,3): error TS2374: Duplicate string index signature.
node_modules/@types/vscode/index.d.ts(4148,3): error TS2374: Duplicate string index signature.
node_modules/@types/vscode/index.d.ts(4573,12): error TS2687: All declarations of 'priority' must have identical modifiers.
node_modules/@types/vscode/index.d.ts(4573,12): error TS2717: Subsequent property declarations must have the same type.  Property 'priority' must be of type 'number', but here has type 'number | undefined'.
node_modules/@types/vscode/index.d.ts(4953,3): error TS2374: Duplicate string index signature.
node_modules/@types/vscode/index.d.ts(6943,3): error TS2717: Subsequent property declarations must have the same type.  Property 'shellArgs' must be of type 'string[] | undefined', but here has type 'string | string[] | undefined'.
node_modules/@types/vscode/index.d.ts(8426,3): error TS2374: Duplicate string index signature.
node_modules/vscode/vscode.d.ts(6,1): error TS6200: Definitions of the following identifiers conflict with those in another file: version, Position, Range, Selection, Keyboard, Mouse, Command, Line, Block, Underline, LineThin, BlockOutline, UnderlineThin, Off, On, Relative, Default, InCenter, InCenterIfOutsideViewport, AtTop, Left, Center, Right, Full, OpenOpen, ClosedClosed, OpenClosed, ClosedOpen, ThemeColor, ThemeIcon, LF, CRLF, Uri, CancellationTokenSource, Disposable, EventEmitter, RelativePattern, GlobPattern, DocumentSelector, ProviderResult, CodeActionKind, CodeAction, CodeLens, DefinitionLink, Definition, Declaration, MarkdownString, MarkedString, Hover, Text, Read, Write, DocumentHighlight, File, Module, Namespace, Package, Class, Method, Property, Field, Constructor, Enum, Interface, Function, Variable, Constant, String, Number, Boolean, Array, Object, Key, Null, EnumMember, Struct, Event, Operator, TypeParameter, SymbolInformation, DocumentSymbol, TextEdit, WorkspaceEdit, SnippetString, ParameterInformation, SignatureInformation, SignatureHelp, Invoke, TriggerCharacter, ContentChange, Unit, Value, Keyword, Snippet, Color, Reference, Folder, CompletionItem, CompletionList, TriggerForIncompleteCompletions, DocumentLink, ColorInformation, ColorPresentation, FoldingRange, Comment, Imports, Region, CharacterPair, None, Indent, IndentOutdent, Outdent, Global, Workspace, WorkspaceFolder, Location, Error, Warning, Information, Hint, DiagnosticRelatedInformation, Unnecessary, Diagnostic, Active, Beside, One, Two, Three, Four, Five, Six, Seven, Eight, Nine, Always, Silent, Never, Shared, Dedicated, New, TaskGroup, ProcessExecution, Escape, Strong, Weak, ShellExecution, Task, taskExecutions, onDidStartTask, onDidEndTask, onDidStartTaskProcess, onDidEndTaskProcess, Unknown, Directory, SymbolicLink, FileSystemError, Changed, Created, Deleted, appName, appRoot, language, clipboard, machineId, sessionId, activeTextEditor, visibleTextEditors, onDidChangeActiveTextEditor, onDidChangeVisibleTextEditors, onDidChangeTextEditorSelection, onDidChangeTextEditorVisibleRanges, onDidChangeTextEditorOptions, onDidChangeTextEditorViewColumn, terminals, activeTerminal, onDidChangeActiveTerminal, onDidOpenTerminal, onDidCloseTerminal, state, onDidChangeWindowState, TreeItem, Collapsed, Expanded, SourceControl, Window, Notification, QuickInputButtons, Manual, AfterDelay, FocusOut, rootPath, workspaceFolders, name, onDidChangeWorkspaceFolders, textDocuments, onDidOpenTextDocument, onDidCloseTextDocument, onDidChangeTextDocument, onWillSaveTextDocument, onDidSaveTextDocument, onDidChangeConfiguration, onDidChangeDiagnostics, inputBox, DebugAdapterExecutable, DebugAdapterServer, DebugAdapterDescriptor, Breakpoint, SourceBreakpoint, FunctionBreakpoint, activeDebugSession, activeDebugConsole, breakpoints, onDidChangeActiveDebugSession, onDidStartDebugSession, onDidReceiveDebugSessionCustomEvent, onDidTerminateDebugSession, onDidChangeBreakpoints, all, onDidChange
node_modules/vscode/vscode.d.ts(4515,12): error TS2687: All declarations of 'priority' must have identical modifiers.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! cat-coding@0.0.1 compile: `tsc -p ./`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the cat-coding@0.0.1 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\kittaakos\AppData\Roaming\npm-cache\_logs\2019-07-10T18_44_47_320Z-debug.log
```

Reply:
I think vscode types are identified as duplicated because you bring them twice. Once via the dependency on `@types/vscode` and the other via the post install script `node ./node_modules/vscode/bin/install` which AFAIK downloads vscode.d.ts to node_modules.

I didn't change the code in vscode-extension-samples at all. Just ran `npm install` and `npm run compile` and I used theia run hosted plugin functionality in order to test it without packaging to vsix.

Reply:
Thank you for the hints, @amiramw.

> used theia run hosted plugin functionality

Can you please help me with this? For instance, [this wiki page](https://github.com/theia-ide/theia/wiki/Testing-VS-Code-Extensions) describes how to try out packaged VS Code extensions. How can I test and verify an extension without packaging it?

Reply:
- Open your vscode extension source code in theia
- Make sure it is compiled and you have the `out` folder
- Run command (F1): "Hosted Plugin: Start Instance"
- Select your repo as the plugin path
- A new tab should be opened for you with theia on port 3030 and with your vscode extension

Reply:
and if you want to just build vsix file it should also be pretty easy and you don't have to edit your package.json:
```
$ npm install -g vsce
$ cd myExtension
$ vsce package
```

(taken from [here](https://code.visualstudio.com/api/working-with-extensions/publishing-extension))

Reply:
Thank you! I am trying this now and will document it if works.

> npm install -g vsce

This is exactly what I wanted to avoid. Probably it is just me, but I hate to install things to the global `npm` 😊

Reply:
You can try [npx](https://github.com/npm/npx). It runs npm packages without installing them.

Reply:
> look for internal server errors after.

@olexii4 did you mean this kind of error? It happens on the `master` (021cfc52d) on macOS anyways:
```
Refused to apply inline style because it violates the following Content Security Policy directive: "default-src 'none'". Either the 'unsafe-inline' keyword, a hash ('sha256-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU='), or a nonce ('nonce-...') is required to enable inline execution. Note also that 'style-src' was not explicitly set, so 'default-src' is used as a fallback.

onLoad @ webviews-main.ts:100
onLoad @ webview.ts:144
(anonymous) @ webview.ts:154
```

Reply:
@kittaakos  No, I did not. I have the next one(on Linux):
```
webview.ts:164 GET http://localhost:3030/webview/home/oleksii/workspace/Microsoft/vscode-extension-samples/webview-sample/media/main.js net::ERR_ABORTED 500 (Internal Server Error)
```

Reply:
I use now FileUri.fsPath which works fine for windows. Could you guys check mac and other OS?